internal/rbd/replicationcontrollerserver.go

```diff
@@ -158,9 +215,65 @@ func (rs *ReplicationServer) EnableVolumeReplication(ctx context.Context,
 			return nil, status.Error(codes.Internal, err.Error())
 		}
 	}
+
+	if interval != "" {
+		err = addSnapshotScheduling(rbdVol, interval, startTime)
+		if err != nil {
+			return nil, err
+		}
+		util.DebugLog(
+			ctx,
+			"Added scheduling at interval %s, start time %s for volume %s",
+			interval,
+			startTime,
+			volumeID)
+	}
+
 	return &replication.EnableVolumeReplicationResponse{}, nil
 }
 
+func addSnapshotScheduling(
+	rbdVol *rbdVolume,
```

Comment:
should `addSnapshotScheduling` maybe be a function of `rbdVolume`?

Reply:
added a  method for rbdImage. this might be helpful for snapshots also